### PR TITLE
feat(create-collector): add test code for create collector

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Things to Talk About

Eslint warns against using await inside the loop. How about ignoring the rules in the test code?

### CODE REVIEW

**Review by yuda110 — APPROVED** [review]
29 steps wow....
I'll listen to the detailed info tomorrow!

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+    test('create collector(1): specific service account Included + schedule ON + Collect Now O', async ({ page }) => {
+        await test.step('1. Access the Asset Inventory > Collector Page.', async () => {
+            await page.goto('/asset-inventory/collector');
+            await expect(await page.getByRole('heading', { name: 'Collector' })).toBeVisible();
+        });
+
+        await test.step('2. Click the [+Create] Button at the top > Verify Successful Move to Step 1/4.', async () => {
+            await page.getByRole('button', { name: 'Create' }).click();
+            await page.mainFrame().waitForURL('/asset-inventory/collector/create');
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:34]: question: what is `page.mainFrame()`? 😮

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+            await expect(page).toHaveURL('/asset-inventory/collector/create');
+        });
+
+        await test.step('3. Search for a keyword in the top search bar and confirm (Check API Call).', async () => {
+            await page.getByPlaceholder('Search', { exact: true }).click();
+            await page.getByPlaceholder('Search', { exact: true }).fill('cloud');
+            await page.getByPlaceholder('Search', { exact: true }).press('Enter');
+            await checkApiCall(page, '/repository/plugin/list');
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:42]: praise: simple!

**Review by piggggggggy — APPROVED** [review]
Real 29 step... LGTM!

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        });
+
+        await test.step('4. Confirm selection of a provider on the left side (Check API Call).', async () => {
+            const providerRadioButtonCount = await page.locator('div.provider > div.p-radio-group.vertical > span').count();
+            const randomIndex = getRandomIndexExceptFirst(providerRadioButtonCount);
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:47]: praise: cool

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            const repositoryRadioButtonCount = await page.locator('div.repository > div.p-radio-group.vertical > span').count();
+            const randomIndex = getRandomIndexExceptFirst(repositoryRadioButtonCount);
+            page.locator(`div.repository > div.p-radio-group.vertical > span:nth-child(${randomIndex + 1})`).click();
+            await checkApiCall(page, '/repository/plugin/list');
+        });
+
+        await test.step('6. Click the [Select] Button for a specific plugin > Confirm Move to Step 2/4.', async () => {
+            await page.locator('div.provider > div.p-radio-group.vertical > span:nth-child(1)').click();
+            await page.locator('div.repository > div.p-radio-group.vertical > span:nth-child(1)').click();
+            await page.waitForLoadState('networkidle');
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:62]: question: Why this statement is needed?

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+            while (isSameName) {
+                collectorTitle = `test collector-${Math.floor(Math.random() * 100)}`;
+                // eslint-disable-next-line no-await-in-loop
+                await page.locator('div.name-form > div.p-text-input').getByRole('textbox').fill(collectorTitle);
+                // eslint-disable-next-line no-await-in-loop
+                isSameName = await checkSameName();
+                if (checkSameNameCount > 10) {
+                    throw new Error('Same name error occurred more than 10 times.');
+                }
+            }
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:109]: question: can you explain what is this logic for? Actually.. it's pretty hard to follow up.😅

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+                const firstServiceAccount = await page.locator('div.p-context-menu.dropdown-context-menu > div.menu-container > span.p-context-menu-item').nth(randomIndex);
+                await firstServiceAccount.click();
+                const firstServiceAccountText = await firstServiceAccount.textContent();
+                if (firstServiceAccountText) selectedServiceAccount.push(firstServiceAccountText);
+                await page.waitForTimeout(2000);
+                serviceAccountCount = await getServiceAccountCount();
+                let randomIndex2 = getRandomIndexInRange(serviceAccountCount);
+                while (randomIndex === randomIndex2) {
+                    randomIndex2 = getRandomIndexInRange(serviceAccountCount);
+                }
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:172]: question: can you explain what is this logic for? 😅

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+                if (selectedPluginName.includes('Prowler Security Compliance Collector')) {
+                    const complianceType = await page.locator('div.collector-options-form-contents form > div:nth-child(2)');
+                    await complianceType.getByRole('button').first().click();
+                    const complianceTypeItemsCount = await complianceType.locator('span.p-context-menu-item').count();
+                    const randomIndex = getRandomIndexInRange(complianceTypeItemsCount);
+                    await complianceType.locator('span.p-context-menu-item').nth(randomIndex).click();
+                }
+            } catch (error) {
+                console.error('error', error);
+            }
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:199]: question: Why catches an error? If there is an error, isn't it right that this case fails?

File: apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+            const scheduleList:number[] = [];
+            for (let i = 0; i < scheduleCount; i++) {
+                // eslint-disable-next-line no-await-in-loop
+                const schedule = await scheduleElement.nth(i).textContent();
+                scheduleList.push(Number(schedule));
+            }
+            await expect(scheduleList.sort()).toEqual((collectorData?.schedule?.hours ?? []).sort());
+        });
+        await test.step('26. Check the Additional Options section: If Additional Options were selected, confirm the chosen or inputted content.', async () => {
+        });
```
Comment (WANZARGEN) [inline apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts:302]: issue: this is empty!

**Review by WANZARGEN — APPROVED** [review]
Oh my.. it must be pretty tough... !!!! What a hard work!
